packages/eslint-config-godaddy-minimal/index.js

```diff
@@ -0,0 +1,24 @@
+module.exports = {
+  extends: [
+    require('./extends')('eslint-config-godaddy-react-flow')
+  ],
+  rules: {
+    'complexity': 0,
+    'consistent-return': 0,
+    'id-length': 0,
+    'indent': [2, 2, {
+      VariableDeclarator: { var: 2, let: 2, const: 3 },
+      flatTernaryExpressions: true,
+      MemberExpression: 'off'
+    }],
+    'max-params': 0,
+    'max-nested-callbacks': 0,
```

Comment:
It's easy enough to turn off the max nested callbacks rule in test folders to prevent the mocha issue where nested callbacks are an OK thing. In regular code, too many nested callbacks is definitely a red flag, so I'd vote to leave this rule in place.

Reply:
If you read my explanation of the rule in the description, this triggers for any nested function invocations, not just for callbacks. This means that using `map` or similar inside an async callback can push code over the count. We all know better than to nest async control flow too deeply, and this can be easily discussed during code review without limiting higher-order functions in general.

Reply:
If an inline map function inside of a deep async control flow puts you over the edge, generally the stuff before it was already too deep. 

You keep mentioning that things can be discussed during code review, but it's very easy to miss certain quality metrics during code reviews, and not all code reviewers have similar value systems when it comes to quality. Having some automatic standards is useful for the parts that are generally detectable, and this is one case, IMO.

Reply:
If code reviewers don't have a good, common set of metrics over quality, then the project in question is destined for poor code quality anyway, and the rest of this is moot. It's not possible to build good code without a common understanding of what "good code" is, and this is a much larger and more complex topic than can be handled by syntactic analysis.

Also, re:
> If an inline map function inside of a deep async control flow puts you over the edge, generally the stuff before it was already too deep.
If that's the case, then the existing rule is already failing to catch problem cases, no? And what about a more complex higher-order function:
```es6
someArray.chain(x => someOtherArray.map(y => [x, y]));
```
This is not a heavy-weight function; it just tuples the cartesian product of two arrays. But putting this in a function cuts the total allowed levels of nesting in half. This is a sign that we're mixing concerns: higher-order functions and callbacks for async control flow are two different issues, that should be addressed differently. Linting, by operating at a syntactic level, is not advanced enough to address these issues adequately. Both of these issues can be addressed by competent devs, so this is not a "something is better than nothing" situation.

Reply:
Doesn't this also discourage more functional programming, where anonymous callbacks are frequently nested? I think this would be a better PR discussion than a linting rule.

Reply:
@LindsayD That's why I run into this problem all the time, despite the fact that I never deeply nest async control flow.